pkg/apis/aadpodidentity/v1/types.go

```diff
@@ -155,7 +155,7 @@ const (
 type AzureIdentityBindingSpec struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 	AzureIdentity     string `json:"azureidentity"`
-	Selector          string `json:"selector"`
+	Selector          map[string]string `json:"selector"`
```

Comment:
Can we use the actual `LabelSelector` type: https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L1076

so that we can get the label from the selector and we can validate user's input:
https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/validation/validation.go#L30